### DISCUSSION TITLE
C4 Plantuml Plugin

### DISCUSSION THREAD

**Discussion by pge76** [docToolchain/docToolchain#949] [Ideas]
Hi there,

first of all, thanks for the awesome work done on this project. I've started using it a few weeks ago, and its awesome.

My Question:
Is is planned to include support for the "unofficial" plantuml Standard Library Repositories to be included?
https://github.com/plantuml-stdlib

I am especially interested in the C4 Model Generation (https://github.com/plantuml-stdlib/C4-PlantUML), as it alignes very well with the arc42 template documentation.

Or maybe, it is already included, and i'm just to stupid to figure out how it works. Then i would appreciated any working sample, how to create such a c4 diagram.

Thanks,
Patrick

**Comment by rdmueller** [comment]
Thanx for the feedback!

How would such a support look like?
As I understand, we need to include some files from plantuml

    !include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml

and it could be nice to have them locally available, correct?

**Reply by pge76** [reply]
yes, that sounds about right. I've got it working with the includes now in my project documentation. But it would be nice, if the includes where already there, and i did not have to specifiy them at every diagram.

**Reply by rdmueller** [reply]
So, a solution could be to magically include all those libs? Will this be possible?

I currently work with live templates in IntelliJ. So I start to type `c4_context` and it will be replaced with the include...
